Avoid DeprecationWarning from sre_compile in Python 3.11

See https://github.com/cpplint/cpplint/pull/214 for discussion.

TODO(jwnimmer-tri) Upstream this to our RobotLocomotion/styleguide.

--- cpplint/cpplint.py
+++ cpplint/cpplint.py
@@ -47,7 +47,7 @@
 import math  # for log
 import os
 import re
-import sre_compile
+import re as sre_compile  # sre_compile is deprecated
 import string
 import sys
 import unicodedata
